src/components/job/JobView.jsx

```diff
@@ -37,12 +38,33 @@ export default function JobView() {
     // this
     return <ErrorInfo />;
   }
-  const { run, data, loading, error } = useRequest(async () => getJob(jobId), {
-    manual: true,
-    onError: () => {
-      toast.error("Failed to get the job information");
+  const { run, data, loading, error } = useRequest(
+    async () => {
+      // return getJob(jobId);
+      return getJob(jobId).then((data) => {
+        if (auth.user.coordinate && data.coordinate) {
+          let userCoordinate = auth.user.coordinate;
+          let jobCoordinate = data.coordinate;
+          const distance = calculateDistance(
+            userCoordinate.latitude,
+            userCoordinate.longitude,
+            jobCoordinate.latitude,
+            jobCoordinate.longitude
+          );
+          data.distance = distance;
+        } else {
+          data.distance = null;
+        }
+        return data;
+      });
     },
-  });
+    {
+      manual: true,
+      onError: () => {
+        toast.error("Failed to get the job information");
```

Comment:
Could be more specific
we may have conditions
- we get the job but not the distance
- we fail to get the job